### DISCUSSION TITLE
Does anyone velcro a laptop to the Create 3?

### DISCUSSION THREAD

**Discussion by tgkell** [iRobotEducation/create3_docs#92] [Questions]
In preparation for working with students in an AI course, I'm currently exploring different Create 3 setups for implementing an autonomous robot that can emulate dog obedience behaviours with voice and hand commands.  To start, I have in mind commands like "come", "stay", virtual roll-over, virtual wag-tail, and maybe even "heel".   Also, I'm open to other ideas.  And maybe this has already been done - is there a recommended way to do it, or an example to follow?

The students will all have a Create 3,  a laptop, many will have a mobile phone, and if it ends up making sense they could be instructed to buy a Pi and Pi camera.  The idea is that I will implement the "robot dog", and then teach the students how to do it with their own Create 3.

For this application I will need a camera on the Create, and one of the possibilities is a camera-equipped Pi 3B or 4B added to the Create, and a wireless laptop would control the Pi.  However, it occurs to me that it might be more effective to reverse that setup, and velcro a laptop with a camera, microphone/speaker, and superior compute power to the Create, use the Pi hand-held (or maybe mobile phone) to communicate with the laptop for high-level control, including a "kill switch".

Laptop sitting on and hardwired to the Create is not one of the "hookup examples", and maybe there's good reason for that.

I'd appreciate hearing opinions on the pros and cons of velcro-ing a laptop to the Create.  Alternative suggestions are welcome too!

**Comment by SamsonHua** [comment]
What applications are you hoping to run on the Raspberry Pi with the camera? If it's basic computer vision or video streaming, I think a Pi should be enough to handle that if you're concerned about computing power. Raspberry Pi 3/4 can also run TFLite models if you need a bit beefier computer vision. 

Not too sure about the reverse setup, how are you planning to interface the Pi if it was hand-held with the laptop? Remote desktop access, VNC, SSH?

**Reply by tgkell** [reply]
What applications?  Good question - I haven't started that part of the process yet, but I expect a Python-based solution with Tensorflow/keras, and/or OpenCV.  I have a feeling that once initial goals are (hopefully) achieved, then more ambitious goals will kick in until some limit is reached, with compute limits being a likely limiting factor.  Anything I'm thinking can be overridden by advice from experienced people.

I'd use SSH over wireless to control the laptop on the Create with the Pi in the hand.  For this kind of development I'm more of a linux person than Windows, so although the students will have Windows laptops, Windows would be running WSL2 Ubuntu, which I have tested with ROS2 and the Create.

The Pi in the hand would really only need a button on GPIO to act as a kill switch, so no need for keyboard/display on the Pi.  Or maybe a few buttons to select from several options in the application on the laptop.  For my purposes (I think) the kill switch is the main requirement for the remote control.  The alternative to a kill switch would be to physically run after the Create and pick it up, but it seems to me a kill switch would come in handy enough to be a requirement.

**Reply by justinIRBT** [reply]
There is no technical reason you can’t use a laptop on top of a create.  The create’s usb is a host, so you may not be able to directly plug into a laptop for ethernet over usb unless it has an OTG port.  You will likely need a usb to ethernet adapter like in the Raspberry Pi 3 hookup example (https://iroboteducation.github.io/create3_docs/hw/hookup/).  We don’t have a suggested way to power a laptop from the robot battery, so you would have to manage that separately.

**Reply by tgkell** [reply]
Since you don't think it's crazy, I like the laptop on Create solution.

But, the usb C to ethernet adapter requirement is noted.  I expect many laptops will not have RJ45, so that's two usb to ethernet adapters, plus an Ethernet cable.  I wonder if straight Ethernet over USB bridge cable might work in some cases.  In the worst case, to save money on adapters, laptop and Create could use wifi, I suppose.

I'm hoping the laptop's own battery will suffice.

Thank you for the replies!
